h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -722,20 +722,28 @@ private FVecParseWriter makeDout(ParseSetup localSetup, int chunkOff, int nchunk
       try {
         switch( cpr ) {
         case NONE:
-          ParserInfo.ParseMethod pm = _parseSetup._parse_type.parseMethod(_keys.length, vec.nChunks());
-          if (pm == ParserInfo.ParseMethod.DistributedParse && ! decryptionTool.isTransparent())
-            pm = ParserInfo.ParseMethod.StreamParse;
+          ParserInfo.ParseMethod pm = _parseSetup.parseMethod(_keys.length, vec);
+          Log.debug("Key " + key + " will be parsed using method " + pm + ".");
+
           if(pm == ParserInfo.ParseMethod.DistributedParse) {
             new DistributedParse(_vg, localSetup, _vecIdStart, chunkStartIdx, this, key, vec.nChunks()).dfork(vec).getResult(false);
             for( int i = 0; i < vec.nChunks(); ++i )
               _chunk2ParseNodeMap[chunkStartIdx + i] = vec.chunkKey(i).home_node().index();
-          } else if(pm == ParserInfo.ParseMethod.StreamParse){
+          } else if(pm == ParserInfo.ParseMethod.StreamParse || pm == ParserInfo.ParseMethod.SequentialParse){
             localSetup = ParserService.INSTANCE.getByInfo(localSetup._parse_type).setupLocal(vec,localSetup);
-            InputStream bvs = vec.openStream(_jobKey);
             Parser p = localSetup.parser(_jobKey);
-            _dout[_lo] = ((FVecParseWriter) p.streamParse(decryptionTool.decryptInputStream(bvs),
-                    makeDout(localSetup,chunkStartIdx,vec.nChunks()))).close(_fs);
+
+            FVecParseWriter writer = makeDout(localSetup,chunkStartIdx,vec.nChunks());
+            final ParseWriter dout;
+            if (pm == ParserInfo.ParseMethod.StreamParse) {
+              InputStream bvs = vec.openStream(_jobKey);
```

Comment:
It is not guaranteed this `InputStream` will be closed in the `streamParse` method one line below. Highly unlikely, but could be caught in a try-with-resources block.

Reply:
Done